engine/src/test/java/com/redhat/cloud/notifications/transformers/BaseTransformerTest.java

```diff
@@ -1,36 +1,153 @@
+    private static final String EVENTS = "events";
+    private static final String METADATA = "metadata";
+    private static final String ORG_ID = "org_id";
+    private static final String PAYLOAD = "payload";
+    private static final String TIMESTAMP = "timestamp";
+
+    /**
+     * Sets up the context and a list with one event for the tests.
+     */
+    @BeforeAll
+    static void setUp() {
+        final Context context = new Context();
+
+        context.setAdditionalProperty(FIXTURE_CONTEXT_ADDITIONAL_PROPERTY, FIXTURE_CONTEXT_ADDITIONAL_PROPERTY_VALUE);
+
+        FIXTURE_CONTEXT = context;
+
+        final List<Event> events = new ArrayList<>();
+        final Event event = new Event();
+
+        final Metadata metadata = new Metadata();
+        metadata.setAdditionalProperty(FIXTURE_METADATA_ADDITIONAL_PROPERTY, FIXTURE_METADATA_ADDITIONAL_PROPERTY_VALUE);
+        event.setMetadata(metadata);
+
+        final Payload payload = new Payload();
+        payload.setAdditionalProperty(FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY, FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY_VALUE);
+        event.setPayload(payload);
+
+        events.add(event);
+        FIXTURE_EVENTS = events;
+    }
 
     @Test
     void shouldContainOrgId() {
         Action action = new Action();
         action.setOrgId("someOrgId");
         action.setTimestamp(LocalDateTime.of(2022, 12, 24, 12, 0));
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        final JsonObject jsonObject = baseTransformer.toJsonObject(action);
 
-        assertEquals("someOrgId", jsonObject.getString("org_id"));
+        assertEquals("someOrgId", jsonObject.getString(ORG_ID));
     }
 
     @Test
     void shouldNotRaiseAnExceptionWhenOrgIdIsMissing() {
         Action action = new Action();
         action.setTimestamp(LocalDateTime.of(2022, 12, 24, 12, 0));
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        final JsonObject jsonObject = baseTransformer.toJsonObject(action);
+
+        assertNull(jsonObject.getString(ORG_ID));
+    }
+
+    /**
+     * Tests that a proper JSON payload is generated from a fully populated {@link Action}.
+     */
+    @Test
+    void toJsonObjectTest() {
+        Action action = new Action();
+
+        action.setAccountId(FIXTURE_ACCOUNT_ID);
+        action.setApplication(FIXTURE_APPLICATION);
+        action.setBundle(FIXTURE_BUNDLE);
+        action.setContext(FIXTURE_CONTEXT);
+        action.setEventType(FIXTURE_EVENT_TYPE);
+        action.setEvents(FIXTURE_EVENTS);
+        action.setOrgId(FIXTURE_ORG_ID);
+        action.setTimestamp(FIXTURE_TIMESTAMP);
+
+        // Call the function under test.
+        JsonObject result = this.baseTransformer.toJsonObject(action);
+
+        // Assert that all the properties from the Action are present in the JSON object.
+        assertTrue(result.containsKey(ACCOUNT_ID), "the account id is missing from the resulting JSON object");
```

Comment:
Is it really necessary to check if a field is missing while there's another assertion below that checks the field value? If the field is missing, then the value check will fail and it will be obvious from JUnit's output that the failure was caused by a missing field.

This is a general question that applies to several "flat" fields in this test but also to several fields in `CamelTypeProcessorTest`. By "flat" fields, I mean field types like `String`, `Boolean`... I do not include fields that contain a JSON structure (`events`, `metadata` or `payload`) in that question.

Reply:
Yeah, I also thought that these tests might be a bit redundant in that sense... I wasn't sure, so I went with the full, verbose way of testing the JSON results. I am going to remove the checks for the primitive types, hang on...

Reply:
I've removed quite a few "primitive key checks" from the tests, and left only the "test the value for this key" tests. The only keys I'm looking for their presence for are the ones that represent another JSON object.